src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -22,13 +22,85 @@ public sealed class TelemetryConfiguration : IDisposable
         private static TelemetryConfiguration active;
 
         private readonly SnapshottingList<ITelemetryInitializer> telemetryInitializers = new SnapshottingList<ITelemetryInitializer>();
+        private ITelemetryChannel telemetryChannel = null;
         private TelemetryProcessorChain telemetryProcessorChain;
         private string instrumentationKey = string.Empty;
         private bool disableTelemetry = false;
         private TelemetryProcessorChainBuilder builder;
         private SnapshottingList<IMetricProcessor> metricProcessors = new SnapshottingList<IMetricProcessor>();
 
         /// <summary>
+        /// Indicates if this instance has been disposed of.
+        /// </summary>
+        private bool isDisposed = false;
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        public TelemetryConfiguration() : this(new InMemoryChannel())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        /// <param name="instrumentationKey">The instrumentation key this configuration instance will provide.</param>
+        public TelemetryConfiguration(string instrumentationKey) : this(new InMemoryChannel())
+        {
+            if (instrumentationKey == null)
+            {
+                throw new ArgumentNullException("instrumentationKey");
+            }
+
+            this.instrumentationKey = instrumentationKey;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        /// <param name="channel">The telemetry channel to provide with this configuration instance.</param>
+        public TelemetryConfiguration(ITelemetryChannel channel)
```

Comment:
In that case:
1. Keep the `TelemetryConfiguration(string, ITelemetryChannel)` constructor because it communicates that both ikey and channel are required for a valid configuration.
2. Keep the `TelemetryConfiguration(string)` constructor because it communicates that ikey is required and creates a default channel.
3. Keep the `TelemetryConfiguration()` constructor because it is required for backward compatibility. Consider applying `[EditorBrowseable(Never)]` attribute to discourage users from picking it by default.
4. Remove the `TelemetryConfiguration(ITelemetryChannel)` because it incorrectly implies that the ikey is optional for any channel.